packages/account-v2/src/pages/ConnectedApps/ConnectedApps.tsx

```diff
@@ -1,18 +1,37 @@
-import React from 'react';
-import { useFetchConnectedApps } from '@deriv/api-v2';
+import React, { useCallback, useState } from 'react';
+import { useFetchConnectedApps, useRevokeConnectedApps } from '@deriv/api-v2';
 import { Loader } from '@deriv-com/ui';
+import { ConnectedAppsRevokeModal } from './ConnectedAppsRevokeModal';
 import { ConnectedAppsTable } from './ConnectedAppsTable';
 
 export const ConnectedApps = () => {
     const { data: connectedApps, isLoading } = useFetchConnectedApps();
+    const { mutate: revokeMutate } = useRevokeConnectedApps();
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [selectedAppId, setSelectedAppId] = useState<number | null>(null);
+
+    const handleToggleModal = useCallback((appId: number | null = null) => {
+        setIsModalOpen(isModalOpen => !isModalOpen);
+        setSelectedAppId(appId);
```

Comment:
shouldnt we set appId before opening the modal

Reply:
it still captures appId even after opening modal 🤔 . What would be the reason for setting it before? 🙏

Reply:
just checking since it may be better to set the updated appid and open the modal